components/brave_wallet/browser/internal/orchard_shard_tree_manager.h

```diff
@@ -0,0 +1,47 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_ORCHARD_SHARD_TREE_MANAGER_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_ORCHARD_SHARD_TREE_MANAGER_H_
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/types/expected.h"
+#include "brave/components/brave_wallet/browser/internal/orchard_block_scanner.h"
+#include "brave/components/brave_wallet/browser/zcash/rust/orchard_shard_tree.h"
+#include "brave/components/brave_wallet/common/zcash_utils.h"
+
+namespace brave_wallet {
+
+class OrchardShardTreeManager {
+ public:
+  OrchardShardTreeManager(
+      std::unique_ptr<::brave_wallet::orchard::OrchardShardTree> shard_tree);
+  ~OrchardShardTreeManager();
+  bool InsertCommitments(OrchardBlockScanner::Result commitments);
```

Comment:
const &

Reply:
We don't want to copy OrchardBlockScanner::Result since it contains OrchardDecodedBlocksBundleImpl  which wraps rust object inside. Changed to r-value instead of reference.